h/traversal/roots.py

```diff
@@ -75,7 +75,14 @@
 from h.traversal import contexts
 
 
-class Root:
+class RootFactory:
```

Comment:
```suggestion
class Base:
```

or `BaseRoot`? I don't think this class is a factory in any sense other than every Python class is a factory for instances of itself (but then you'd have to call every single class `FooFactory`...)

(Pyramid calls these root factories because they can be any callable, so you can pass a standalone factory function or something, but we're not doing that)

Reply:
It's a factory in the Pyramid Root Factory sense, not in a general programming sense.

Almost every docstring in this file starts with: `Root factory for...` so this seems like the right name to me.